src/components/GenreSelect.js

```diff
@@ -0,0 +1,21 @@
+import '../styles/GenreSelectStyles.css';
+
+export default function GenreSelect({genreNames, selectedGenre, onSelect}) {
+    return (<div className="GenreSelect">
+        {genreNames.map((genreName, i) => {
+            if (genreName === selectedGenre) {
+                return (<span className={"GenreSelect__item selected"}
+                              key={i}
+                              onClick={() => onSelect(genreName)}>
+                    {genreName}
+                </span>)
+            } else {
+                return (<span className={"GenreSelect__item"}
+                              key={i}
+                              onClick={() => onSelect(genreName)}>
+                    {genreName}
+                </span>)
```

Comment:
here you repeat the logic twice. Only selected classname distinguish two returns. Make cleaner, please. 
![image](https://github.com/user-attachments/assets/ae603c12-6242-461a-b7c3-783e6cf74748)

And moreover when I choose another genre I see it in the console, but styling underlining do not change.